status-im/status-react
# Feature: Implement import of key-pair with recovery phrase inside wallet settings

[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #20037 

### Summary

* This PR attempts to implement the flow for importing a key-pair by entering the recovery/seed phrase from inside the wallet settings.
* This PR also attempts to refactor the common component for entering the recovery-phrase.
  * The common component has been refactored to share most of the logic while supporting a custom form title, navigation icon, and submit button.

### Notes

* Currently the toast for displaying the confirmation after importing a key-pair is not themed correctly in the video.
  * Waiting on https://github.com/status-im/status-mobile/pull/20002 to merged to a get a fix for this.

### Testing Notes

While manual testing this PR for merge:
* The key pairs and accounts settings are feature-flagged and can be manually tested when the feature flag is removed.
* Onboarding while recovering an account with a seed-phrase could be affected, so it would be good to quickly check that it's still working. Testing has happened during development, and the feature seemed to be working, but it would be nice to double-check. 🙏

#### Platforms


- Android
- iOS

#### Areas that maybe impacted


##### Functional

- Key Pairs and Accounts settings in Wallet Settings
- Onboarding with recovering account with seed-phrase
- Using import seed-phrase from Wallet while creating account

### Steps to test


To test these changes there are some requirements, the user account for testing will need two devices, and both of those devices must be synced to the user account.

Once the account and devices are setup to sync between the devices:

* Open the Status app (mobile or desktop) on one device (device-a) and create a new account with a key-pair.
Choose to create a new account from the wallet home screen, then select to create a new key-pair by generating a recovery phrase key-pair, and then finish creating the account with that new key-pair.
Because the devices are synced, the second device (device-b) should be updated with the knowledge of the new account and the key-pair.

* On the second device (device-b), navigate to the mobile wallet settings and open the "Key pairs and accounts" settings. At the top of screen there should be a section for missing key-pairs, and that section should display the current missing key-pair and the icon emojis for the related accounts.

* Press on the actions menu icon ("...") for the missing key-pair and then press on the "Import by entering recovery phrase" action. Proceed by entering the recovery-phrase and confirming the form submission by authenticating.

* If successful, the user should be navigated back to the key-pairs and accounts settings page, and they should see a toast confirming their success, and they should see the imported key-pair in the list of key-pairs.

* If unsuccessful, the user should be navigated back to the recovery-phrase screen and should be displayed the related error with their submission.



### Before and after screenshots comparison


https://github.com/status-im/status-mobile/assets/2845768/94af4209-0586-4221-ab34-e475824ad7f8



status: ready
